### PR TITLE
feat: Create Auth prompt and add examples

### PR DESCRIPTION
**Pull request by rajat-sr**

### CODE REVIEW

File: lib/types/auth.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const FormPrompt = require('../prompts/form');
+
+const defaultVerify = () => {
```
Comment (doowb) [inline lib/types/auth.js]: Please change `defaultVerify` to `defaultAuthenticate`.

File: lib/types/auth.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    constructor(options) {
+      super(options);
+    }
+
+    async submit() {
+      this.value = await verify.call(this, this.values, this.state);
+      super.base.submit.call(this);
+    }
+
+    static create(verify) {
```
Comment (doowb) [inline lib/types/auth.js]: Please change `verify` to `authenticate`.

File: lib/prompts/basicauth.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      }
+    }
+  ];
+
+  class BasicAuthPrompt extends AuthPrompt.create(verify) {
+    constructor(options) {
+      super({ ...options, choices });
+    }
+
+    static create(verify) {
```
Comment (doowb) [inline lib/prompts/basicauth.js]: Please change `verify` to `authenticate`.

File: lib/prompts/basicauth.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const AuthPrompt = require('../types/auth');
+
+function defaultVerify(value, state) {
+  if (value.username === this.options.username && value.password === this.options.password) {
+    return true;
+  }
+  return false;
+}
+
+const factory = (verify = defaultVerify) => {
```
Comment (doowb) [inline lib/prompts/basicauth.js]: Please change `verify` to `authenticate`.

File: lib/prompts/basicauth.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const AuthPrompt = require('../types/auth');
+
+function defaultVerify(value, state) {
```
Comment (doowb) [inline lib/prompts/basicauth.js]: Please change `defaultVerify` to `defaultAuthenticate`.

File: examples/auth/totp.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const crypto = require('crypto');
+const base32 = require('hi-base32');
+const AuthPrompt = require('../../lib/types/auth');
+
+async function verify(value, state) {
```
Comment (doowb) [inline examples/auth/totp.js]: Please change `verify` to `authenticate`.

**Comment by rajat-sr** [comment]
Authenticate is definitely a better name. I've made the requested changes.

**Comment by doowb** [comment]
Thanks for the update!

**Pull request merged**